repairnator/src/main/java/fr/inria/spirals/repairnator/process/step/GatherTestInformation.java

```diff
@@ -162,6 +163,17 @@ public FileVisitResult preVisitDirectory(Path file, BasicFileAttributes attrs) t
         	this.shouldStop = true;
         }
         
-        this.shouldStop = (inspector.getMode() == RepairMode.FORBEARS) ? !this.shouldStop : this.shouldStop;
+        if (this.inspector instanceof ProjectInspector4Bears) {
+        	if (!((ProjectInspector4Bears)this.inspector).isAboutAPreviousBuild()) {
```

Comment:
It's over complicated: here you need a specific behaviour when gathering test info on passing build, so why not extending this class instead of playing with instance of?